code/modules/hydroponics/grown/replicapod.dm

```diff
@@ -133,11 +142,13 @@
 				if(isobserver(M))
 					var/mob/dead/observer/O = M
 					if(O.ckey == ckey && O.can_reenter_corpse)
-						make_podman = TRUE
+						if(ghost_consent(M))
+							make_podman = TRUE
```

Comment:
I'm worried about the UX on this, the player harvesting it would have to click the plant, where nothing obvious is happening, because the proc is sleeping (waiting for user input on the ghost's side) 

This means they're probably gonna spam-click the plant and in turn spam the ghost with popups
(Which is also very likely to cause bugs, I don't think you have any sanity checks for this) 

As an idea for a replacement, maybe there can be a screen alert on the ghost's top right, saying "your ghost is in a replica pod! Click this to revive when it is harvested (or unclick it to not revive) (or maybe even 'alt-click this to dismiss')"